c2rust-refactor/scripts/upgrade_ptr_to_ref.lua

```diff
@@ -567,24 +795,36 @@ function Visitor:rewrite_call_expr(expr)
     if segments and segments[#segments] == "free" and first_param_expr:get_kind() == "Cast" then
```

Comment:
I think it's because it was so common to see free called on a non void pointer and so the translation would add the implicit cast to a void pointer.

I suppose this could be improved to not require the param to be a cast, but it is certainly the common case